<img src="docs/genaclust_logo.jpg" alt="logo" title="GKclust logo" height="400" align="right" />

# GKclust

GKclust is a computational tool designed to detect clusters of mutations in complete genomes. 

## Installation

To install GKclust from GitHub navigate to a directory where you want to install the program and clone the repository:
```
git clone https://github.com/KazanovLab/GKclust
```

To compile GKclust and install it system-wide navigate into the cloned directory and run:
```
cd GKclust
make
```

## Quick start

GKclust can generate output in several formats simultaneously.
* The short output (-s option) includes only mutations that are part of the identified clusters.
* The full output (-f option) contains all mutations from the input VCF files.
* The list of cluster (-c option) contains only list of clusters for particular VCF file.
* The brief statistics (-a option) for all processed VCFs.

Other options include:
* The input directory containing VCF file (-i) or the list of full paths to VCF files (-p) in a separate text file.
* The reference genome location (-g).
* The output directory (-o).
* The p-value for determining the between-mutation distance threshold (-t, default: 0.01).

Examples:

Short format output:
```
gkclust -s -i /inputdir/ -o /outputdir/ -g /refgenome/hg19.fa
```

Full format output:
```
gkclust -f -i /inputdir/ -o /outputdir/ -g /refgenome/hg19.fa
```


## Reporting Bugs and Feature Requests
Please use the [GitHub issue tracker](https://github.com/KazanovLab/GENAKITclust/issues) to report bugs or suggest features.

## Citing
Ponomarev GV, KurtoÄŸlu B, Kazanov MD. "GKclust: A Bioinformatics Tool for Detecting Spatially Clustered Mutations in Complete Genomes". *to be submitted*

## Funding
This study was supported by the Scientific and Technological Research Council of Turkey (TUBITAK) under Grant Number 123E476. The authors thank TUBITAK for their support. 

